ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -2814,6 +2820,80 @@ MmCheckSystemImage(IN HANDLE ImageHandle,
     return Status;
 }
 
+
+PVOID
+NTAPI
+LdrpFetchAddressOfSecurityCookie(PVOID BaseAddress, ULONG SizeOfImage)
+{
+    PIMAGE_LOAD_CONFIG_DIRECTORY ConfigDir;
+    ULONG DirSize;
+    PVOID Cookie = NULL;
+
+    /* Check NT header first */
+    if (!RtlImageNtHeader(BaseAddress)) return NULL;
+
+    /* Get the pointer to the config directory */
+    ConfigDir = RtlImageDirectoryEntryToData(BaseAddress,
+        TRUE,
+        IMAGE_DIRECTORY_ENTRY_LOAD_CONFIG,
+        &DirSize);
+
+    /* Check for sanity */
+    if (!ConfigDir ||
+        (DirSize != 64 && ConfigDir->Size != DirSize) ||
+        (ConfigDir->Size < 0x48))
+        return NULL;
```

Comment:
```suggestion
    if (!ConfigDir ||
        (DirSize != 64 && ConfigDir->Size != DirSize) ||
        (ConfigDir->Size < 0x48))
    {
        return NULL;
    }
```

Reply:
Also, 0x48 == 72== `sizeof(IMAGE_LOAD_CONFIG_DIRECTORY)` for its 32-bit version (this is by the way platform specific, with `IMAGE_LOAD_CONFIG_DIRECTORY32` for 32-bits and `IMAGE_LOAD_CONFIG_DIRECTORY64` for 64 bits).

Reply:
And 64 == `FIELD_OFFSET(IMAGE_LOAD_CONFIG_DIRECTORY, SecurityCookie)`

Reply:
See also https://git.reactos.org/?p=reactos.git;a=blob;f=modules/rostests/apitests/loadconfig/common.c;hb=68182b2a52858eb0011df1467444e66e0b056f74#l47 that does a similar thing.
And these magic values should also be fixed in this function: https://git.reactos.org/?p=reactos.git;a=blob;f=dll/ntdll/ldr/ldrinit.c;hb=68182b2a52858eb0011df1467444e66e0b056f74#l410